### ISSUE TITLE
What's the purpose of using scale_factor?

### ISSUE THREAD

**Issue by TPreece101** [Nixtla/mlforecast#29]
I noticed in the docs under the "Custom predictions" section it references using a `scale_factor` - I'm just wondering what the purpose of this would be?

Is it the same purpose as the `alpha` here?: https://www.kaggle.com/code/lemuz90/m5-mlforecast/notebook 

I'm assuming that it's some kind of post prediction adjustment to improve accuracy but I'm keen to hear the thought process behind it.

**Comment by TPreece101** [comment]
Ah OK interesting. So every time that `fcst.predict(28, my_predict_fn, alpha=alpha)` calls the `my_predict_fn` the output of that is added to the features for predicting the next timestep?

How would you go about finding some good values for this? Would it just be via cross validation?

Is there any logic behind them being greater than 1?

**Comment by jmoralez** [comment]
> So every time that fcst.predict(28, my_predict_fn, alpha=alpha) calls the my_predict_fn the output of that is added to the features for predicting the next timestep?

Yes, internally the target values are updated each time with the new prediction, so when a feature needs the value of the target past the training set the predictions are used as if they were the future values of the target.

> How would you go about finding some good values for this? Would it just be via cross validation?

I haven't done this anywhere else but that'd be a good way.

> Is there any logic behind them being greater than 1?

I think the reason is that people found the model was mostly underestimating so they wanted to increment the predictions a bit.

**Comment by kdgutier** [comment]
The M5 competition has a downward bias in the trend in the test set.
If you multiply the predictions of a good performing algorithm in validation by 0.95 you will get misguiding good results.

**Comment by TPreece101** [comment]
Ah sorry I forgot to reply to this, thank you for your answers, this makes sense. I've started using the library and I love it so far - I have starred the repo 😊
